[15.0][MIG] sale_order_line_input: Migration to 15.0

cesar-tecnativa opened the pull request:
@Tecnativa TT36521

carlosdauden committed:
[ADD] sale_order_line_input: New module

[UPD] README.rst

[UPD] Update sale_order_line_input.pot

Commit:
[MIG] sale_order_line_input: migration to v12.0

rven committed:
[MIG] sale_order_line_input: Migration to 12.0

[UPD] Update sale_order_line_input.pot

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_order_line_input
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_order_line_input/

marylla committed:
Added translation using Weblate (German)

[UPD] README.rst

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (17 of 17 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_order_line_input
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_order_line_input/zh_CN/

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_order_line_input
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_order_line_input/

Rad0van committed:
[IMP] sale_order_line_input: black, isort, prettier

Rad0van committed:
[MIG] sale_order_line_input: Migration to 13.0

[UPD] Update sale_order_line_input.pot

[UPD] README.rst

gurneyalex committed:
[FIX] sale_order_line_input: bump version

to solve a pypi upload issue

CarlosRoca13 committed:
[FIX] sale_order_line_input: The field customer does not exist

Since v13 the customer field on res.partner model has been disappeared,
so when we try to add a value on the field order_partner_id we get an
error.

sale_order_line_input 13.0.1.1.0

carlosdauden committed:
[FIX] sale_order_line_input: company_id field is empty and multi-company domains don't work. Taxes empty

bealdav committed:
[MIG] sale_order_line_input: Migration to 14.0

[MIG] sale_order_line_input: Migration to 14.0

bealdav committed:
IMP sale_order_line_input: add write_date on view

[UPD] Update sale_order_line_input.pot

[UPD] README.rst

[IMP] update dotfiles [ci skip]

[UPD] README.rst

[UPD] README.rst

cesar-tecnativa commented:
@CarlosRoca13 could you check?

cesar-tecnativa commented:
@pedrobaeza could you check please?

pedrobaeza commented on sale_order_line_input/models/sale_order.py:
```diff
@@ -11,45 +11,45 @@ class SaleOrderLine(models.Model):
         readonly=True,
     )
-    force_company_id = fields.Many2one(
+    with_company_id = fields.Many2one(
```
You don't need to rename this field.

pedrobaeza commented on sale_order_line_input/models/sale_order.py:
```diff
@@ -0,0 +1,67 @@
+        """Create order to correct compute of taxes"""
+        if not self.order_partner_id or self.order_id:
+            return
+        SaleOrder = self.with_context(with_company=self.with_company_id.id).env[
```
https://www.odoo.com/documentation/15.0/es/developer/howtos/company.html

```suggestion
        SaleOrder = self.with_company(self.force_company_id).env[
```

pedrobaeza commented:
/ocabot migration sale_order_line_input

pedrobaeza commented on sale_order_line_input/models/sale_order.py:
```diff
@@ -27,7 +27,7 @@ def _compute_force_company_id(self):
                 line.order_id.company_id
                 # Is not necessary use browse here
-                or self.env.context.get("force_company")
+                or self.env.context.get("with_company")
```
I think you have to look into `allowed_company_ids` context variable.

pedrobaeza replied:
This is not correct afaik. You should check allowed companies

carlosdauden commented on sale_order_line_input/i18n/es.po:
```diff
@@ -0,0 +1,136 @@
+#: model_terms:ir.ui.view,arch_db:sale_order_line_input.view_sales_order_line_input_tree
+#, fuzzy
+msgid "Open Sale Order"
+msgstr "Línea de pedido de venta"
```
Bad translation

carlosdauden commented on sale_order_line_input/views/sale_order_line_view.xml:
```diff
@@ -0,0 +1,214 @@
+                />
+                <button
+                    name="action_sale_order_form"
+                    string="Open Sale Order"
```
Remove this to avoid waste space

francesco-ooops committed:
Added translation using Weblate (Italian)

SicurSam committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (24 of 24 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_order_line_input
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_order_line_input/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (24 of 24 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_order_line_input
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_order_line_input/it/

victoralmau committed:
[IMP] sale_order_line_input: black, isort, prettier

victoralmau committed:
[MIG] sale_order_line_input: Migration to 15.0

TT36521

victoralmau commented:
Could you review it again?

sergio-teruel commented:
@victoralmau Add yourself as contributor

sergio-teruel commented:
ping @carlosdauden

carlosdauden commented:
Attend this:
https://github.com/OCA/sale-workflow/pull/2114#discussion_r930477479

victoralmau commented:
Changes done.

sergio-teruel commented:
@pedrobaeza review, if all is ok, Can you merge it?

carlosdauden commented on sale_order_line_input/models/sale_order.py:
```diff
@@ -0,0 +1,62 @@
+            line.force_company_id = (
+                line.order_id.company_id
+                # Is not necessary use browse here
+                or self.env.context.get("with_company")
```
Don't change original line:
```suggestion
                or self.env.context.get("force_company")
```

victoralmau replied:
Change done.

pedrobaeza commented:
/ocabot merge nobump
